### DISCUSSION TITLE
How to align the AI-deck image frame and imu data ?

### DISCUSSION THREAD

**Discussion by coo122567** [bitcraze/discussions#1466] [Q&A]
Hello everyone!
How can I align the frame with the imu data (pitch、roll、yaw、x、y、z) ? I have tried running fpv.py and found that the imu data and the frame cannot be aligned. The frame rate is much slower than the imu rate. Is there any way to align the imu with the frame? I hope to align the imu with the frame! Please help, thank you!

**Comment by evoggy** [comment]
Hi,

Yes this is probably possible but requires some work for implementing. The easiest might be to use [CPX](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/cpx/) for synchronizing the time when the frame has been acquired directly on the Crazyflie. Please have a look at the GAP8/STM32 CPX example [located here](https://github.com/bitcraze/crazyflie-firmware/tree/master/examples/app_stm_gap8_cpx) to get started.
